src/ctia/entity/relationship.clj

```diff
@@ -62,6 +71,69 @@
   (st/merge PagingParams
             RelationshipFieldsParam))
 
+(s/defschema IncidentCasebookLinkRequest
+  {:casebook_id Reference
+   (s/optional-key :tlp) TLP})
+
+(def incident-casebook-link-route
+  (POST "/:id/link" []
+        :return rs/Relationship
+        :body [link-req IncidentCasebookLinkRequest
+               {:description "an Incident Link request"}]
+        :header-params [{Authorization :- (s/maybe s/Str) nil}]
+        :summary "Link an Incident to a Casebook"
+        :path-params [id :- s/Str]
+        :capabilities #{:create-incident
+                        :create-relationship
+                        :read-casebook}
+        :auth-identity identity
+        :identity-map identity-map
+        (let [incident (read-store :incident
+                                   read-record
+                                   id
+                                   identity-map
+                                   {})
+              casebook (read-store :casebook
+                                   read-record
+                                   (-> link-req
+                                       :casebook_id
+                                       long-id->id
+                                       :short-id)
+                                   identity-map
+                                   {})]
+
+          (cond
+            (not incident)
+            (not-found)
+            (not casebook)
+            (not-found)
```

Comment:
IMO, if the casebook is not found, it's a bad request. The `not-found` response is only related to the request URI.